src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -293,44 +292,55 @@ public BlockHeaderModel GetBlockHeader(string hash, bool isJsonFormat = true)
         }
 
         /// <summary>
-        /// Returns information about a bitcoin address
+        /// Returns information about a bitcoin address and it's validity.
         /// </summary>
-        /// <param name="address">bech32 or base58 BitcoinAddress to validate.</param>
-        /// <returns>ValidatedAddress containing a boolean indicating address validity</returns>
-        /// <exception cref="ArgumentNullException">Thrown if address provided is null/empty.</exception>
+        /// <param name="address">The bech32 or base58 <see cref="BitcoinAddress"/> to validate.</param>
+        /// <returns><see cref="ValidatedAddress"/> instance containing information about the bitcoin address and it's validity.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if address provided is null or empty.</exception>
         [ActionName("validateaddress")]
         [ActionDescription("Returns information about a bech32 or base58 bitcoin address")]
         public ValidatedAddress ValidateAddress(string address)
         {
-            if (string.IsNullOrEmpty(address))
-                throw new ArgumentNullException("address");
+            Guard.NotEmpty(address, nameof(address));
 
-            var res = new ValidatedAddress();
-            res.IsValid = false;
+            var result = new ValidatedAddress
+            {
+                IsValid = false,
+                Address = address,
+            };
 
             // P2WPKH
             if (BitcoinWitPubKeyAddress.IsValid(address, this.Network, out Exception _))
             {
-                res.IsValid = true;
+                result.IsValid = true;
             }
-            // P2WSH
+            // P2WSH (unsupported)
             else if (BitcoinWitScriptAddress.IsValid(address, this.Network, out Exception _))
             {
-                // We don't support P2WSH addresses yet
-                res.IsValid = false;
+                result.IsValid = false;
```

Comment:
I'm not so sure about this `false`. Whether we support them from a wallet perspective or not, the address format can surely be determined to be valid. It may be network-specific as to whether the address format is acceptable, dependent on soft fork deployment status etc.